src/component/modal.js

```diff
@@ -34,15 +34,39 @@ const styles = StyleSheet.create({
   },
 });
 
-const Modal = ({ title = '', onClose, children, style }) => (
-  <View style={[styles.modal, style]}>
-    <H4Text style={styles.title}>{title.toUpperCase()}</H4Text>
-    <Button style={styles.cancelBtn} onPress={onClose}>
-      <Icon image="cancel-grey" style={styles.cancelIcon} />
-    </Button>
-    {children}
-  </View>
-);
+class Modal extends Component {
+  constructor(props) {
+    props.title = props.title || '';
+    super(props);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyDown);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  handleKeyDown(event) {
+    if (event.keyCode === 27) {
+      this.props.onClose();
+    }
+  }
+
+  render() {
+    const { title, style, onClose, children } = this.props;
+    return (
+      <View style={[styles.modal, style]}>
+        <H4Text style={styles.title}>{title.toUpperCase()}</H4Text>
+        <Button style={styles.cancelBtn} onPress={onClose}>
+          <Icon image="cancel-grey" style={styles.cancelIcon} />
+        </Button>
+        {children}
+      </View>
+    );
+  }
+}
```

Comment:
I'm wondering if we should pull this out into an `Escapable` component that the modal inherits from.